lib/Core/ErrorState.cpp

```diff
@@ -329,6 +322,25 @@ ref<Expr> ErrorState::propagateError(Executor *executor,
     valueErrorMap[instr] = result;
     return result;
   }
+  case llvm::Instruction::GetElementPtr: {
+    // Check if base is symbolic and whether a symbolic error value already
+    // exists for the newly pointed element
+    ref<Expr> newError;
+    if (isInStoredError(result)) {
+      newError = retrieveStoredError(result);
```

Comment:
@Himeshi Returning `newError` here is incorrect, because what is returned is the error of the value stored at the pointer address. Here what we want is to return the error in the pointer. In the handling of `llvm::Instruction::GetElementPtr`, I recommend instead that we use the old structure as follows to compute the error value computed for the gep. 
```
    // Simply propagate error of the first argument
    ref<Expr> error = arguments.at(0).error;
    if (error.isNull()) {
      error = ConstantExpr::create(0, Expr::Int8);
      llvm::Value *v = instr->getOperand(0);
      std::map<llvm::Value *, ref<Expr> >::iterator it = valueErrorMap.find(v);
      if (it != valueErrorMap.end()) {
        error = valueErrorMap[v];
      }
      if (error->getWidth() > Expr::Int8)
        error = ExtractExpr::create(error, 0, Expr::Int8);
    }
    valueErrorMap[instr] = error;
```
However, we add a new check to generate the new error variable for the value that is *stored*, roughly as you have implemented, with slight modifications.
```
ref<Expr> storedError = retrieveStoreError(result);
if (storedError.isNull()) {
    ref<Expr> baseError = retrieveStoredError(arguments[0].value);
    // The following also performs nullity check on the result of the cast. If the type does not match, re is NULL
    if (ReadExpr *re = llvm::dyn_cast<ReadExpr>(baseError)) {
      std::string errorName = re->updates.root->name;
      const Array *newErrorArray =
          errorArrayCache.CreateArray(errorName, Expr::Int8);
      UpdateList ul(newErrorArray, 0);
      ref<Expr> offset = SubExpr::create(result, arguments[0].value);
      newError = ReadExpr::create(ul, offset);
      executeStoreSimple(instr, result, newError);
  }
}
```
Note that the above does not compute the return error value, it simply computes the error for the value stored at the result address.
And finally we return the correct error:
```
return error;
```